tapeagents/environment.py

```diff
@@ -203,3 +223,38 @@ def close(self) -> None:
 
     def last_actions(self, tape: Tape) -> list[Action]:
         return [step for step in tape.steps[-tape.metadata.n_added_steps :] if isinstance(step, Action)]
+
+    async def areact(self, tape: Tape) -> Tape:
+        for action in self.last_actions(tape):
+            t = time.perf_counter()
+            observation = await self.astep(action)
+            observation.metadata.other["action_execution_time"] = time.perf_counter() - t
```

Comment:
Should this metadata be included in the non-async version?

Reply:
we have it in the sync version of "def step()"

Reply:
Nitpick: should the async and sync versions follow the same pattern? In this case then the metadata would be added in astep instead?

Reply:
Yes, definitely